codegen/src/codegen/config/config.py

```diff
@@ -0,0 +1,147 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import dataclasses
+import enum
+import textwrap
+from typing import List
+
+import humps
+
+
+@enum.unique
+class FieldType(enum.Enum):
+    Nothing = enum.auto()
+    Bool = enum.auto()
+    Int = enum.auto()
+    Long = enum.auto()
+    Float = enum.auto()
+    Double = enum.auto()
+    String = enum.auto()
+
+
+@enum.unique
+class RefinedFieldType(enum.Enum):
+    Nothing = enum.auto()
+    RustUsize = enum.auto()
+
+
+@dataclasses.dataclass
+class ConfigField:
+    name: str
+    desc: str
+
+    ty: FieldType = FieldType.Nothing
+    refined_ty: RefinedFieldType = RefinedFieldType.Nothing
+
+    required: bool = False
+    sensitive: bool = False
+
+    # default value for the config field if not set
+    default: str = ""
+    # a list of available values for the config field
+    available: List[str] = dataclasses.field(default_factory=list)
+    # a list of example values for the config field
+    example: List[str] = dataclasses.field(default_factory=list)
+
+    ###########
+    # RustGen #
+    ###########
+    def rust_type(self) -> str:
```

Comment:
I don't know what's the idiom to group methods. In Golang, receiver methods can scatter in multiple files. In Python, perhaps we need some design pattern like "XlangConfig(Field)Generator" and take "Config(Field)" a parameter.